src/components/shared/NoPermissionMessage/NoPermissionMessage.tsx

```diff
@@ -22,7 +22,7 @@ const MSG = defineMessages({
 
 interface Props {
   requiredPermissions: ColonyRole[];
-  domainName?: string;
+  domainName?: string | null;
```

Comment:
Can this ever be null? If `domain.metadata` is present, the name will be there
<img width="337" alt="image" src="https://user-images.githubusercontent.com/112586815/224106811-52f80ae4-13f7-4de7-8cab-bf63cede0242.png">

Reply:
It can be, in this case inside `TransferFundsDialogForm`:
```
          <NoPermissionMessage
            requiredPermissions={[ColonyRole.Funding]}
            domainName={
              hasRoleInFromDomain
                ? fromDomain?.metadata?.name ||
                  `Domain #${fromDomain?.nativeId}`
                : toDomain?.metadata?.name || `Domain #${toDomain?.nativeId}`
            }
          />
```

`fromDomain` and `toDomain` come from here:
```
  const fromDomain = colonyDomains.find(
    (domain) => domain?.nativeId === fromDomainId,
  );
```

```
  const toDomain = colonyDomains.find(
    (domain) => domain?.nativeId === toDomainId,
  );
```

So unfortunately even if `.metadata` wasn't optional, the `.find` assumes that it could be `null` in the case that it doesn't find the domain.

Reply:
I don't think that's the case (although I might be wrong). If `.find` doesn't find anything, it'll return `undefined` instead of `null`. Removing `null` from the prop type doesn't trigger any type errors

Reply:
You are right, it does return `undefined`. I do remember adding the `null` because at some point Typescript was yelling me about it :thinking: But that doesn't seem to be case for `TransferFundsDialogForm` anymore